### ISSUE TITLE
fix(ml): derive the augmentation block from the decided target geometry and the open scale question

### ISSUE THREAD

**Issue by LukeSantossz** [LukeSantossz/visiosoil-app#195]
## Description

The `augmentation` block in `ml/config.yaml` predates the 2026-08-11 answers and ADR 0014, and two of its settings now contradict what those records establish about the target and about deployment. Neither value is derived from a decision and neither is marked provisional against the issue that will decide it.

## Context

Found by the architecture review of 2026-08-25.

The block was written when the target's shape was unknown and the deployment scale was assumed comparable to the collection scale. ADR 0014 settled the first and opened the second. The values did not move.

**Scale.** `zoom_range: [0.95, 1.05]` is ±5 % scale jitter. ADR 0014 records that the dataset's millimetres per pixel is constant by construction and the application's is unknown and variable, and lists scale-invariant training as one of three candidate resolutions. Whichever way #184 decides, ±5 % is not that resolution and it is not a derived number. The defect is not that the value is wrong — nobody knows yet — it is that a load-bearing hypothesis is sitting in a config file with no comment saying it is one, in a repository whose other config values carry their reasoning inline.

**Symmetry.** `vertical_flip: false` and `rotation_range: 15`. The target is an isotropic disc photographed top-down, so it has no canonical orientation: the eight elements of the square's symmetry group — four rotations of 90° and their reflections — are **exactly** label-preserving and create no pixel the camera did not record. An arbitrary 15° rotation is the lossy member of the same family: it exposes corners that were never photographed and must fill them. Keras `RandomRotation` fills per `fill_mode`, which the config does not set, so the default applies and the corners carry fabricated content — on a task whose label *is* texture statistics. The pipeline is using the lossy form of a transform whose lossless form is available, free, and unused.

This is not the same finding as #180. That is about the sampling filter; this is about which transforms the block declares.

## Current State

`ml/config.yaml`:

```yaml
augmentation:
  horizontal_flip: true
  vertical_flip: false
  rotation_range: 15
  brightness_range: [0.85, 1.15]
  contrast_range: [0.9, 1.1]
  zoom_range: [0.95, 1.05]
  translation_range: 0.05
```

Realized in `ml/src/preprocess.py:58-128`. `RandomFlip("horizontal")` at `:76`, `RandomFlip("vertical")` at `:79` gated off, `RandomRotation(rotation/360.0)` at `:83` with no `fill_mode` argument, `RandomZoom(height_factor=...)` at `:116-118`.

`ml/src/config.py:160-202` validates the ranges thoroughly — ascending bounds, the `RandomBrightness` domain, the `RandomContrast` symmetry constraint — and every one of those checks is about whether Keras can express the value. Nothing checks whether the value follows from a decision, which is the right division of labour; the missing half is a comment, not a validator.

By contrast, `training.deterministic_ops` at `ml/config.yaml:60-64` and `data.datasets_dir` at `:16-19` both carry their reasoning inline and name the record behind them. The augmentation block carries none.

## Proposed Solution

Two changes, one mechanical and one editorial, plus an experiment arm.

1. **Add the symmetry augmentation.** Set `vertical_flip: true`, and add a `rot90: true` option realized as a random `k` in `tf.image.rot90` in the augmentation path. There is no Keras layer for it, and the three-line map function is preferable to `RandomRotation(factor=1.0)`, which is arbitrary-angle and reintroduces fill. Keep `rotation_range` as a separate, smaller jitter if E13 shows it earns its fill artefacts.
2. **Comment every value with the decision it derives from, or mark it provisional and name the issue that will decide it.** `zoom_range` gets the second treatment, pointing at #184. This is the convention the rest of the file already follows.
3. **Add the symmetry arm to E13** in `soil-classification.md` §11. E13 is already the ROI-shape experiment and the two interact: 90° rotations are exact under a centred square and under a circular mask, and under the inscribed square they are exact as well, so the arm is a free rider on an experiment already scheduled rather than a new one.

Criteria for this shape: the eightfold group is exact rather than approximate, so it multiplies the training set with no artefact and no label risk, which is the acceptance test `data-augmentation.md` gives for any augmentation; the comment convention is the file's own; and folding the measurement into E13 costs one arm on an experiment that is already comparing preprocessing variants over one dataset.

**One caution on ordering.** `rot90` is exact only over a square input, so it belongs after the centred-square crop that SPEC 0035 introduces to `ml/src/preprocess.py`, not before. Adding it to the current whole-frame path would rotate a non-square image and reintroduce exactly the fill it exists to avoid.

## Dependencies

- **SPEC 0035 / #79** — the crop that makes `rot90` exact.
- **#184** — decides what `zoom_range` should be. This issue does not pre-empt that; it marks the value as awaiting it.
- Related to the scale-simulation issue, which replaces `zoom_range` as the mechanism for scale variation if E14 supports it.
- Related to #183, since the symmetry arm's effect has to clear the detectable-effect floor to be adoptable on evidence rather than on argument.

## Acceptance Criteria

- [ ] `vertical_flip` is enabled and a `rot90` option exists, validated by `load_config` and realized in the augmentation path after the centred-square crop.
- [ ] A test asserts that the eightfold symmetry augmentation produces outputs that are exact permutations of the input pixels — no interpolation, no fill.
- [ ] Every value in the `augmentation` block carries either the decision it derives from or a provisional marker naming the issue that will decide it. `zoom_range` names #184.
- [ ] E13 in §11 carries the symmetry arm.
- [ ] The record states that `rot90` requires a square input and therefore depends on SPEC 0035.

**Comment by LukeSantossz** [comment]
**The open question this issue defers to is now closed, and it resolves `zoom_range` in a stronger direction than "not derived".**

This issue records that `zoom_range: [0.95, 1.05]` is a load-bearing hypothesis sitting in a config file, and correctly declines to call it wrong because #184 had not decided how scale would be handled. #184 is closed. [ADR 0017](../blob/main/docs/adr/0017-scale-is-read-by-a-classical-operator-on-a-known-circle.md) reads the scale from an object of known size and refuses when there is none, and [SPEC 0037](../blob/main/docs/specs/0037-scale-normalised-greyscale-patch-pipeline.md) normalises every photograph to one canonical millimetres-per-pixel before the model sees anything. Scale invariance was **not** the resolution chosen; measurement was.

So the verdict on this value changes. It is not an undocumented hypothesis any more — it is a transform that undoes the pipeline's own normalisation, on a task whose label is defined by absolute particle size. It belongs in the same category this issue already puts `RandomRotation` in, and in the same category the transfer-learning literature puts mirroring a traffic sign: **a transform strong enough to be useful is strong enough to change the class the laboratory measured.** ±5 % is small, but the axis is the wrong one in principle, and a small wrong-axis jitter is worse than an honest one because nothing about the result looks broken.

The remedy is now a removal rather than a comment: drop `zoom_range` from the block, and record in its place that scale is measured and normalised, not augmented over. If a scale-robustness margin is ever wanted, it belongs downstream of the normalisation and against a measured tolerance, not as a blind ±5 % on the raw frame.

**One thing this does not settle**, and it is why this comment does not just close the item: the normalisation is one-directional and driven by the coarsest kept photograph, so the training set already contains a real, measured scale spread. Whether the model needs any jitter on top of that is an empirical question, and it is the kind of question #204 now owns — the pretrained backbone arrives scale-invariant by construction, which is the opposite problem.

The symmetry half of this issue is unaffected and is corroborated from the transfer side: soil in a dish has no canonical orientation, so the flip invariance ImageNet builds in costs nothing here, unlike the traffic-sign case. `vertical_flip: true` and exact 90° rotations remain free and unused.
